python/simulation.py

```diff
@@ -3169,3 +3169,52 @@ def quiet(quietval=True):
 
 def verbosity(verbose_val):
     mp.cvar.verbosity = verbose_val
+
+def merge_subgroup_data(data,num_groups):
+    # Lazy import
+    from mpi4py import MPI    
+    
+    # Pull required parameters
+    comm = MPI.COMM_WORLD
+    num_workers = comm.Get_size()
+    shape=data.shape
+    size = data.size
+    out_shape = shape + (num_groups,)
+    dtype=data.dtype.char
+    
+    # Initialize new input and output datasets
+    input=np.zeros(shape,dtype,order='F')
+    input[:]=data
+    output=np.zeros(out_shape,dtype,order='F')
+    
+    # Specify how much talking each proc will do
+    count_in = np.array([size] * num_workers)
+    count_out = np.array([size] * num_workers)
```

Comment:
So it looks like all of the workers are sending data, but multiple workers overwrite the same section of the output array?   Shouldn't we just send from `myrank() == 0` in each group?

Reply:
I had trouble preventing deadlocks, which is why I originally did it this way. I've overcome that issue though. Now only group masters send their data, but every process receives data.